### ISSUE TITLE
jQuery syntax for complex object doesn't work on HTTP GET

### ISSUE THREAD

**Issue by gdoron** [aspnet/Mvc#5123]
$ = jQuery

```
$.param({Paragraphs:[{Text:"text",
                    Title:"title", Id:1}]})
```

Produces:

> Paragraphs%5B0%5D%5BText%5D=text&Paragraphs%5B0%5D%5BTitle%5D=title

```
decodeURIComponent("Paragraphs%5B0%5D%5BText%5D=text&Paragraphs%5B0%5D%5BTitle%5D=title"): // "Paragraphs[0][Text]=text&Paragraphs[0][Title]=title"
```

This doesn't work on GET.
But If I change from

"Paragraphs[0][Text]=text&Paragraphs[0][Title]=title"
To
"Paragraphs[0].Text=text&Paragraphs[0].Title=title"
It does work.

Worth checking if there's a standard regarding how to send complex objects over HTTP GET request.

**Comment by rynowak** [comment]
As discussed in another thread, this is in the query string not form data. We currently have support for the jQuery syntax using form data. Should we add this for query string as well?

@gdoron if you want to unblock your usage of this you'd need to create something like [this](https://github.com/aspnet/Mvc/blob/dev/src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/JQueryFormValueProviderFactory.cs) - but use the query string data rather than form data. Then add to `MvcOptions.ValueProviderFactories`

**Comment by gdoron** [comment]
@rynowak If there's no overhead of adding this notation for querystring globally, please do.
jQuery uses this notation, and I believe all other web framework do too,

If there's a significant overhead, maybe adding an attribute to the model?

	public ActionResut Preview([JqueryNotation] model){}
	
I'm just concerned that .NET have a different standard from the rest of the world.

**Comment by rynowak** [comment]
Closing this as a duplicate of https://github.com/aspnet/Mvc/issues/6361 - which we are planning to make on by default in 2.1.0 -
